src/CommunityToolkit.Maui.Markup.SourceGenerators/SourceGenerators/TextAlignmentExtensionsGenerator.cs

```diff
@@ -1,377 +1,375 @@
-
-	public void Initialize(IncrementalGeneratorInitializationContext context)
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+
+namespace CommunityToolkit.Maui.Markup.SourceGenerators;
+
+[Generator(LanguageNames.CSharp)]
+class TextAlignmentExtensionsGenerator : IIncrementalGenerator
+{
+	const string iTextAlignmentInterface = "Microsoft.Maui.ITextAlignment";
+	const string mauiControlsAssembly = "Microsoft.Maui.Controls";
+
+	public void Initialize(IncrementalGeneratorInitializationContext context)
 	{
 		// Get All Classes in User Library
-		var userGeneratedClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
-			static (syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax { BaseList: not null },
-			static (context, cancellationToken) =>
-			{
-				var compilation = context.SemanticModel.Compilation;
-
-				var iTextAlignmentInterfaceSymbol = compilation.GetTypeByMetadataName(iTextAlignmentInterface);
-
-				if (iTextAlignmentInterfaceSymbol is null)
-				{
-					throw new Exception("There's no .NET MAUI referenced in the project.");
-				}
-
-				var classSymbol = (INamedTypeSymbol?)context.SemanticModel.GetDeclaredSymbol(context.Node);
-
-				// If the ClassDlecarationSyntax doesn't implements those interfaces we just return null
-				if (classSymbol?.AllInterfaces.Contains(iTextAlignmentInterfaceSymbol, SymbolEqualityComparer.Default) is not true)
-				{
-					return null;
-				}
-
-				return classSymbol;
-			});
+		var userGeneratedClassesProvider = context.SyntaxProvider.CreateSyntaxProvider(
+			static (syntaxNode, cancellationToken) => syntaxNode is ClassDeclarationSyntax { BaseList: not null },
+			static (context, cancellationToken) =>
+			{
+				var compilation = context.SemanticModel.Compilation;
+
+				var iTextAlignmentInterfaceSymbol = compilation.GetTypeByMetadataName(iTextAlignmentInterface);
+
+				if (iTextAlignmentInterfaceSymbol is null)
+				{
+					throw new Exception("There's no .NET MAUI referenced in the project.");
+				}
+
+				var classSymbol = context.SemanticModel.GetDeclaredSymbol((ClassDeclarationSyntax)context.Node, cancellationToken);
+				if (classSymbol is null || classSymbol.DeclaringSyntaxReferences[0].GetSyntax(cancellationToken) != context.Node)
+				{
+					// In case of multiple partial declarations, we want to run only once.
+					// So we run only for the first syntax reference.
+					return null;
+				}
+
+				while (classSymbol is not null)
+				{
+					if (classSymbol.ContainingAssembly.Name == mauiControlsAssembly)
+					{
+						break;
+					}
+
+					if (classSymbol.Interfaces.Any(i => i.Equals(iTextAlignmentInterfaceSymbol, SymbolEqualityComparer.Default) || i.AllInterfaces.Contains(iTextAlignmentInterfaceSymbol, SymbolEqualityComparer.Default)))

```

Comment:
maybe it's a copy/paste issue... But you're checking against the same symbol twice

Reply:
@pictos `i.Equals(iTextAlignmentInterfaceSymbol, SymbolEqualityComparer.Default)` checks exactly the current interface, while `i.AllInterfaces.Contains(iTextAlignmentInterfaceSymbol, SymbolEqualityComparer.Default)` checks all interfaces it implements.

`classSymbol.AllInterfaces` won't play nice here as it will include interfaces from base types that could come from maui. I'm looping through base types and break once we get into maui.

Per my understanding, this was the intention of the deduplication code that existed before. But correct me if I'm misunderstanding. It would also be good to support any wrong behavior with an additional test case.

Reply:
Not sure about that, I'll need to see what's beeing generated and if that match to what we want... I don't have time to dig into that during this week. @brminnick if you've time let me know otherwise we can wait until the weekend (?)

Reply:
Thanks @pictos! If you’re able to take a look this weekend, that’d be awesome! I’m jumping on a plane to London tomorrow and wont have time to dive into it until I get back in ~2 weeks

Reply:
@brminnick have a safe travel ❤️